geode-management/src/main/java/org/apache/geode/cache/configuration/RegionConfig.java

```diff
@@ -467,6 +478,28 @@ public String getId() {
     return getName();
   }
 
+  @Override
+  public boolean equals(Object that) {
+    if (that == null) {
+      return false;
+    }
+
+    if (this == that) {
+      return true;
+    }
+
+    if (!(that instanceof RegionConfig)) {
+      return false;
+    }
+
+    RegionConfig config = (RegionConfig) that;
+
+    if (this.getId().equals(config.getId()) && this.getType().equals(config.getType())) {
+      return true;
+    }
+
+    return false;
+  }
```

Comment:
can use IntelliJ to generate a much more succinct equals method, but functionally this looks solid